### PR TITLE
Juanro/refactoring constructors

### PR DESCRIPTION
**Pull request by Juanro-Minne**
Refactor AuthRoleResultDto and chartDto, clientDto 

All related Unit test are passing!

image
image
image

### CODE REVIEW

File: RGO.App.Tests/Controllers/HRIS/ChartControllerUnitTests.cs [hunk 1]
```diff
@@ -149,13 +154,15 @@ public async Task UpdateChartData_ReturnsNotFound_OnException()
...
         var labels = new List<string> { "Label1", "Label2", "Label3" };
         var data = new List<int> { 10, 20, 30 };
-        var mockChartDto = new ChartDto(
-                                        1,
-                                        "Disabilities",
-                                        "bar",
-                                        dataTypes,
-                                        labels,
-                                        data);
+        var mockChartDto = new  ChartDto
```
Comment (EstiaanBritz) [inline RGO.App.Tests/Controllers/HRIS/ChartControllerUnitTests.cs:157]:
```suggestion

        var mockChartDto = new  ChartDto
```

File: RGO.App.Tests/Controllers/HRIS/ChartControllerUnitTests.cs [hunk 2]
```diff
@@ -149,13 +154,15 @@ public async Task UpdateChartData_ReturnsNotFound_OnException()
...
+                            {
+                                Id = 1,
+                                Name = "Disabilities",
+                                Type = "bar",
+                                DataTypes = dataTypes,
+                                Labels = labels,
+                                Data = data
+                            };
```
Comment (EstiaanBritz) [inline RGO.App.Tests/Controllers/HRIS/ChartControllerUnitTests.cs:165]:
```suggestion
                            };
                            
```

File: RGO.UnitOfWork.Tests/Entities/ChartUnitTests.cs [hunk 3]
```diff
@@ -16,30 +16,35 @@ public void chartTest()
...
+            Id = 0,
+            Name = "Name",
+            Type = "Type",
+            DataTypes = new List<string> { "data 1", "data 2" },
+            Labels = new List<string> { "Label1", "Label2" },
+            Data = new List<int> { 1, 2 }
+        };
+
+        return new Chart(chartDto);
+
```
Comment (ReinoRabbit) [inline RGO.UnitOfWork.Tests/Entities/ChartUnitTests.cs:30]: Space

Comment (Juanro-Minne) [reply RGO.UnitOfWork.Tests/Entities/ChartUnitTests.cs:30]: done

**Review by ReinoRabbit — APPROVED** [review]

**Review by DewanAlberts — APPROVED** [review]

File: RGO.Services.Tests/Services/ChartServiceUnitTests.cs [hunk 4]
```diff
@@ -229,12 +233,16 @@ public async Task GetChartDataTest()
     public async Task DeleteChartTest()
     {
         var chartId = 1;
-        var expectedChartDto = new ChartDto(chartId,
-                                            "Test",
-                                            "Pie",
-                                            new List<string> { "Gender", "Race" },
-                                            new List<string> { "Male", "Female" },
-                                            new List<int> { 1, 1 });
+        var expectedChartDto = new ChartDto
```
Comment (EstiaanBritz) [inline RGO.Services.Tests/Services/ChartServiceUnitTests.cs:236]:
```suggestion

        var expectedChartDto = new ChartDto
```

File: RGO.Services.Tests/Services/ClientServiceUnitTest.cs [hunk 5]
```diff
@@ -16,10 +17,11 @@ public ClientServiceUnitTest()
     {
         _dbMock = new Mock<IUnitOfWork>();
         _clientService = new ClientService(_dbMock.Object);
-        _clientDto = new ClientDto(
-                                   1,
-                                   "string"
-                                  );
+        _clientDto = new ClientDto
```
Comment (EstiaanBritz) [inline RGO.Services.Tests/Services/ClientServiceUnitTest.cs:20]:
```suggestion

        _clientDto = new ClientDto
```

File: RGO.UnitOfWork.Tests/Entities/ClientUnitTest.cs [hunk 6]
```diff
@@ -17,10 +17,10 @@ public void ClientTest()
     [Fact]
     public async Task ClientToDtoTest()
     {
-        var clientDto = new ClientDto(1, "Name");
+        var clientDto = new ClientDto{ Id = 1, Name = "Name" };
         var client = new Client(clientDto);
         var dto = client.ToDto();
         Assert.Equal(dto.Id, client.Id);
```
Comment (EstiaanBritz) [inline RGO.UnitOfWork.Tests/Entities/ClientUnitTest.cs:23]:
```suggestion

        Assert.Equal(dto.Id, client.Id);
```

**Review by ReinoRabbit — APPROVED** [review]

**Review by DewanAlberts — APPROVED** [review]

**Review by EstiaanBritz — APPROVED** [review]

**Review by ensanga — APPROVED** [review]

**Review by ensanga — APPROVED** [review]

**Pull request merged**